cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -97,12 +112,51 @@ func main() {
 
 				if len(unschedulablePodsToHelp) == 0 {
 					glog.V(1).Info("No unschedulable pods")
-					continue
+				} else {
+					scaledUp, err := ScaleUp(unschedulablePodsToHelp, nodes, migConfigs, gceManager, kubeClient, predicateChecker)
+					if err != nil {
+						glog.Errorf("Failed to scale up: %v", err)
+					} else {
+						if scaledUp {
+							lastScaleUpTime = time.Now()
+							// No scale down in this iteration.
+							continue
+						}
+					}
 				}
 
-				_, err = ScaleUp(unschedulablePodsToHelp, nodes, migConfigs, gceManager, kubeClient, predicateChecker)
-				if err != nil {
-					glog.Errorf("Failed to scale up: %v", err)
+				if *scaleDownEnabled {
+
+					// In dry run only utilization is updated
+					calculateUtilizationOnly := lastScaleUpTime.Add(*scaleDownDelay).After(time.Now()) ||
+						lastScaleDownFailedTrial.Add(*scaleDownTrialFrequency).After(time.Now())
+
+					allScheduled, err := scheduledPodLister.List()
+					if err != nil {
+						glog.Errorf("Failed to list scheduled pods: %v", err)
+						continue
+					}
+					underutilizedNodes = CalculateUnderutilization(
+						nodes,
+						underutilizedNodes,
+						*scaleDownUtilizationThreshold,
+						allScheduled)
+
+					if !calculateUtilizationOnly {
+						result, err := ScaleDown(
+							nodes,
+							underutilizedNodes,
+							*scaleDownUnderutilizationTime,
+							allScheduled,
+							gceManager, kubeClient)
+						if err != nil {
+							glog.Errorf("Failed to scale down: %v", err)
+						} else {
+							if result == ScaleDownNodeDeleted {
```

Comment:
IIUC ScaleDownNodeDeleted mean successfully removed so the statement should be different here.